src/Avalonia.NameGenerator/Domain/ControlType.cs

```diff
@@ -0,0 +1,4 @@
+﻿namespace Avalonia.NameGenerator.Domain
+{
+    internal record ControlType(string Name, string FullName, bool isWindow);
```

Comment:
Notably, `ControlType.FullName` and `ResolvedView.ClassName` are holding the same data now for no reason. Worth removing the `ControlType` record completely. We already have all the neccessary info inside `ResolvedView`, and hense we can perform the `is window` check right inside the `InitializeComponentCodeGenerator` by comparing `ResolvedView.ClassName` with `Avalonia.Controls.Window`

Reply:
Ah my bad, the `ResolvedView.ClassName` holds the name of the control itself, not the base class.